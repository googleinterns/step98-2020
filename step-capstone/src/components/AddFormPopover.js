import React from 'react';
import { Typography, Popover, Card, CardActions, CardContent, Button, TextField } from '@material-ui/core';
import AddForm from './AddForm'

export default function AddFormPopver(props) {
    return (
        <div>
            <Popover
                id={props.id}
                open={props.open}
                anchorEl={props.anchorEl}
                onClose={props.onClose}
                anchorOrigin={props.anchorOrigin}
                transformOrigin={props.transformOrigin}
            >
                <AddForm onClose={props.onClose}/>
            </Popover>
        </div>
    )
}