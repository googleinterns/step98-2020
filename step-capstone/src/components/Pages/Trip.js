import React from 'react';
import Finalized from '../Sidebars/Finalized';
import Unfinalized from '../Sidebars/Unfinalized';
import AddItemButton from '../TravelObjectForms/AddItemButton'
import { Grid, Box } from '@material-ui/core'
import '../../styles/Trip.css'
import { FirebaseContext } from '../Firebase';
import MapComponent from "../Utilities/Map"
import { handleSuggestions } from "../../scripts/Suggestions"
import GetSuggestionButton from '../Utilities/GetSuggestionButton';

const config = {
    userCategories: ["bakery"],
    userBudget: 4,
    radius: "10000",
    timeRange: [new Date(2020, 7, 15, 2, 0), new Date(2020, 7, 15, 20, 0)],
    coordinates: { lat: 36.1699, lng: -115.1398 },
    items: new Set([])
}
//"ChIJmx1Uvc3FyIARdp6ftqC7Gd8","ChIJDRyBe_nEyIARH77JCHU27r8","ChIJVcuReVnbyIARXwAzHqxeAnk","ChIJ_YX8a9LGyIARrcojBX4AgtU"
export default class Trip extends React.Component {
    static contextType = FirebaseContext;
    constructor(props) {
        super(props);

        this.state = {
            reference: "/users/" + sessionStorage.getItem("userId") + "/trips/" + sessionStorage.getItem("tripId"),
            items: null,
            tripSetting: null,
            selectedObject: null,
            today: {
                events: [],
                date: null
            },
            map: null,
            service: null,
            queryResults: null
        }

        this.handleRemoveItem = this.handleRemoveItem.bind(this);
        this.handleEditItem = this.handleEditItem.bind(this);
        this.handleAddItem = this.handleAddItem.bind(this);
        this.handleEditTripSetting = this.handleEditTripSetting.bind(this);
        this.handleSelectedObject = this.handleSelectedObject.bind(this);
        this.handleChangeDisplayDate = this.handleChangeDisplayDate.bind(this);
        this.setMap = this.setMap.bind(this);
    }

    componentDidMount() {
        let travelObjectList = [];
        this.context.getTrip(this.state.reference)
            .then(data => {
                let trip = data.data();
                this.setState({
                    tripSetting: {
                        title: trip.title,
                        startDate: trip.startDate.toDate(),
                        endDate: trip.startDate.toDate(),
                        destination: trip.destination,
                        description: trip.description,
                        userPref: trip.userPref
                    }
                })

                trip.travelObjects.forEach(travelObject => {
                    travelObject.startDate = travelObject.startDate.toDate();
                    travelObject.endDate = travelObject.endDate.toDate();
                    travelObjectList.push(travelObject)
                });
                this.setState({ items: travelObjectList });
            })
            .catch(error => {
                console.log("Error retrieving trip data");
                console.error(error);
            });
    }

    handleRemoveItem(data) {
        this.context.deleteTravelObject(this.state.reference, data)
            .then(() => {
                this.setState({
                    items: this.state.items.filter((item) => item.id !== data.id)
                });
            })
            .catch(error => {
                console.log("Error Removing Item")
                console.error(error)
            });
    }

    handleEditItem(data) {
        let newItems = [];
        let itemToChange;
        this.state.items.forEach((item) => {
            if (item.id === data.id) {
                itemToChange = item;
                newItems.push(data);
            } else {
                newItems.push(item);
            }
        });
        this.context.editTravelObject(this.state.reference, itemToChange, data)
            .then(() => {
                this.setState({
                    items: newItems,
                    loaded: true
                });
            })
            .catch((error) => {
                console.log("Error Editing Item");
                console.log(error);
            });
        this.setState({ items: newItems });
    }

    handleAddItem(data) {
        // Add to database here
        data.id = Date.now();
        this.context.addTravelObject(this.state.reference, data)
            .then(() => {
                this.setState({ items: this.state.items.concat(data) });
                this.getFoodSuggestions(config).then(results => {
                    console.log(results)
                });
            })
            .catch(error => {
                console.log("Error Adding Item")
                console.error(error)
            });
    }

    async handleEditTripSetting(newSetting) {
        await this.context.editTripSetting(this.state.reference, this.state.tripSetting, newSetting);
        this.setState({
            tripSetting: newSetting
        });
    }

    handleSelectedObject(id) {
        this.setState({ selectedObject: id })
    }

    handleChangeDisplayDate(travelObjects, date) {
        if (this.state.today.date !== date) {
            this.setState({
                today: {
                    events: travelObjects,
                    date: date
                }
            })
        }
    }

    async getActivitySuggestions(config) {
        /**
         * param: 
         * config: an object with six fields: 
         *  1. userCategories: a  String array of categories 
         *  2. userBudget: an integer for budget
         *  3. radius: a string integer radius object 
         *  4. timeRange: free time range [startDate, endDate]
         *  5. coordinates: an object for coordinates
         *  6. items: set of all place ids of selected places
         * return the suggestions : an array of PlaceObject already sorted based on score
         */
        if (this.state.map) {
            let results = await handleSuggestions(this.state.service, config, "activities");
            return results;
        }
    }

    async getFoodSuggestions(config) {
        if (this.state.map) {

            let results = await handleSuggestions(this.state.service, config, "food");
            return results;
        }
    }

    setMap(map) {
        this.setState({
            map: map,
            service: new window.google.maps.places.PlacesService(map)
        })
    }

    render() {
        // if data hasn't been loaded yet, don't render the trip
        // prevents map from loading empty data
        if (!this.state.items || !this.state.tripSetting) {
            return null;
        }
        return (
            <div className="trip">
                <Grid id="map-component">
                    <MapComponent
                        zoom={15}
                        defaultCenter={this.state.tripSetting.destination.coordinates}
                        finalized={this.state.items.filter((item) => item.finalized)}
                        unfinalized={this.state.items.filter((item) => !item.finalized && item.coordinates !== null)}
                        selected={this.state.selectedObject}
                        displayDate={this.state.today}
                        setMap={this.setMap}
                    />
                </Grid>
                <Grid container className="foreground" direction="row" justify="space-between">
                    <Grid item id="finalized-component">
                        <Finalized
                            list={this.state.items.filter((item) => item.finalized)}
                            startDate={this.state.tripSetting.startDate}
                            endDate={this.state.tripSetting.endDate}
                            onRemoveItem={this.handleRemoveItem}
                            onEditItem={this.handleEditItem}
                            onAddItem={this.handleAddItem}
                            onClickObject={this.handleSelectedObject}
                            setTodaysEvents={this.handleChangeDisplayDate}

                        />
                    </Grid>
                    <Grid item id="unfinalized-component">
                        <Unfinalized
                            list={this.state.items.filter((item) => !item.finalized)}
                            onRemoveItem={this.handleRemoveItem}
                            onEditItem={this.handleEditItem}
                            onAddItem={this.handleAddItem}
                            tripSetting={this.state.tripSetting}
                            onEditTripSetting={this.handleEditTripSetting}
                            onClickObject={this.handleSelectedObject}
                        />
                    </Grid>
                </Grid>
                <Grid id="button-group">
                    <Box mb={3}>
                        <GetSuggestionButton />
                    </Box>
                    <Box>
                        <AddItemButton
                            startDate={this.state.tripSetting.startDate}
                            onAddItem={this.handleAddItem}
                        />
                    </Box>
                </Grid>
            </div>
        );
    }
}