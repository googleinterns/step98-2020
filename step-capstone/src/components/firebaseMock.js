import React from 'react';
//Firebase Mock Setup 

const londonTestTripData = {
  description: "A trip to London",
  endDate: new Date (2020, 7, 10),
  location: "London",
  startDate: new Date (2020, 7, 7),
  title : "London Trip",
  travelObjects : {
    arrivalAirport: "LON",
    arrivalDate: new Date(2020, 7, 7, 8),
    departureAirport: "MCO",
    departureDate: new Date (2020, 7, 7, 15),
    description: "flight from Orlando to London",
    finalized: false,
    id: 0,
    type: "flight"
  }
}

const londonTestTrip = {
  data : () => {return londonTestTripData}
}

const testTrips = [londonTestTrip]

class firebaseMock {
  getTripList() {
    return new Promise((resolve) => {
      resolve(testTrips);
    });
  }
  getTrip() {
    return new Promise ((resolve) => {
      resolve(londonTestTrip);
    })
  }
}
export default firebaseMock;