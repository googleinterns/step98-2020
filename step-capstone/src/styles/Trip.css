.test {
    background-color: blueviolet;
}

#map-component {
    height: 100%;
    width: 100%;
    position: absolute;
    z-index: 0;
    background-color: royalblue;
}

#finalized-component {
    width: 300px;
    background-color: red;
    height: 100%;
    overflow-y: auto;
    pointer-events: auto;
    box-shadow: 1px 1px 7px gray;
}

#unfinalized-component {
    width: 300px;
    background-color: red;
    height: 100%;
    overflow-y: auto;
    pointer-events: auto;
    box-shadow: 1px 1px 7px gray;
}

#trip {
    height: 100vmin;
    position: relative;
}

.foreground {
    width: 100%;
    height: 100vmin;
    position: absolute;
    z-index: 2;
    pointer-events: none;
}

#add-button-component {
    z-index: 5;
    position: absolute;
    bottom: 0;
    right: 0;
    margin: 30px;
}

#add-form-container {
    height: 500px;
    width: 600px;;
}