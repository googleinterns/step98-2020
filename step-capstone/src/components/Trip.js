import React from 'react';
import TravelObject from './TravelObject'
import AddItemButton from './AddItemButton'
import { Grid } from '@material-ui/core'
import '../styles/Trip.css'
import Map from "./Map"
import {FirebaseContext} from './Firebase';

export default class Trip extends React.Component {
  static contextType = FirebaseContext;
    constructor(props) {
        super(props);

        this.state = {
            reference :"/users/" + sessionStorage.getItem("userId") +"/trips/" + sessionStorage.getItem("tripId"),
            items: []
        }

        this.handleRemoveItem = this.handleRemoveItem.bind(this);
        this.handleEditItem = this.handleEditItem.bind(this);
        this.handleAddItem = this.handleAddItem.bind(this);
    }

    componentDidMount() {
      let travelObjectList = [];
      this.context.getData(this.state.reference)
      .then(data => {
        data.data().travelObjects.forEach(travelObject => {
          travelObjectList.push(travelObject)
        });
        this.setState({items : travelObjectList});
      })
      .catch(error => {console.log("Error Getting Trip Data")});
    }

    handleRemoveItem(data) {
      this.context.deleteTravelObject(this.state.reference, data)
      .then(() => {
        this.setState({
          items: this.state.items.filter((item) => item.id !== data.id)
        });
      })
      .catch(error => console.log("Error Removing Item"));
    }

    handleEditItem(data) {
      this.setState({
          items: this.state.items.map((item) => {
              if (item.id === data.id) {
                let success = this.context.editTravelObject(this.state.reference, item, data)
                  .then(() => {
                    return true;
                  })
                  .catch(error => {
                    console.log("Error Editing Item");
                    return false; 
                  });
                return (success ? data : item);
              } else {
                  return item;
              }
          })
      })
    }

    handleAddItem(data) {
        if (data === undefined) {
            console.log("please enter information");
        } else {
            // Add to database here
            data.id = Date.now();
            this.context.addTravelObject(this.state.reference, data)
            .then(() => {
              this.setState({items : this.state.items.concat(data)});
            })
            .catch(error => {console.log("Error Adding Item")});
        }
    }

    render() {
        if (this.props.items === undefined) {
            // TODO: redirect to Home page
        }
        return (
            <div className="trip">
                <Grid id="map-component"><Map zoom={13} center={{ lat: 51.5, lng: 0.087 }} /></Grid>
                <Grid container className="foreground" direction="row" justify="space-between">
                    <Grid item id="finalized-component">
                        <Finalized
                            list={this.state.items.filter((item) => item.finalized)}
                            onRemoveItem={this.handleRemoveItem}
                            onEditItem={this.handleEditItem}
                            onAddItem={this.handleAddItem}
                        />
                    </Grid>
                    <Grid item id="unfinalized-component">
                        <Unfinalized
                            list={this.state.items.filter((item) => !item.finalized)}
                            onRemoveItem={this.handleRemoveItem}
                            onEditItem={this.handleEditItem}
                            onAddItem={this.handleAddItem}
                        />
                    </Grid>
                </Grid>
                <Grid id="add-button-component">
                    <AddItemButton onAddItem={this.handleAddItem} />
                </Grid>
            </div>
        );
    }
}

/*
 Classes for testing UI purposes
*/
function Finalized(props) {
    return (
        <Grid>
            {
                props.list.map((item) => {
                    return <TravelObject
                        key={item.id}
                        data={item}
                        onRemoveItem={props.onRemoveItem}
                        onEditItem={props.onEditItem}
                        onAddItem={props.onAddItem}
                    />
                })
            }
        </Grid>
    )
}

function Unfinalized(props) {
    return (
        <Grid>
            {
                props.list.map((item) => {
                    return <TravelObject
                        key={item.id}
                        data={item}
                        onRemoveItem={props.onRemoveItem}
                        onEditItem={props.onEditItem}
                        onAddItem={props.handleAddItem}
                    />
                })
            }
        </Grid>
    )
}
