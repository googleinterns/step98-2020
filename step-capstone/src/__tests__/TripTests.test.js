import React from 'react';
import Trip from '../components/Trip';
import renderer from 'react-test-renderer';
import {FirebaseContext } from '../components/Firebase';
import firebaseMock from '../components/firebaseMock'

test('Trip displays correctly', () => {
  sessionStorage.setItem("userId", "testUser");
  sessionStorage.setItem("tripId", "testTrip");
  
  const tripComponent = renderer.create(
    <FirebaseContext.Provider value={new firebaseMock()}>
      <Trip/>
    </FirebaseContext.Provider>
    
  );

  let test = tripComponent.toJSON();
  expect(test).toMatchSnapshot();
});