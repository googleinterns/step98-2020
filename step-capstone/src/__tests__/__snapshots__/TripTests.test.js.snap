// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trip displays correctly 1`] = `
<div
  className="trip"
>
  <div
    className="MuiGrid-root"
    id="map-component"
  >
    <div
      id="map"
    />
  </div>
  <div
    className="MuiGrid-root foreground MuiGrid-container MuiGrid-justify-xs-space-between"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
      id="finalized-component"
    >
      <div>
        <div>
          <div>
            <header
              className="header"
            >
              <div
                className="calendar__title"
              >
                <div
                  className="icon secondary chevron_left"
                  onClick={[Function]}
                >
                  ‹
                </div>
                <button
                  className="secondary"
                >
                  EST
                </button>
                <div
                  className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
                  style={
                    Object {
                      "left": "15px",
                      "maxWidth": "141px",
                      "position": "relative",
                    }
                  }
                >
                  <label
                    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink={false}
                    htmlFor="displayDate"
                    id="displayDate-label"
                  >
                    Display Date
                  </label>
                  <div
                    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      autoFocus={false}
                      className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                      disabled={false}
                      id="displayDate"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="text"
                      value="07/14/2020"
                    />
                    <div
                      className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                    >
                      <button
                        className="MuiButtonBase-root MuiIconButton-root"
                        disabled={false}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onDragLeave={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        tabIndex={0}
                        type="button"
                      >
                        <span
                          className="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden={true}
                            className="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                            />
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                          </svg>
                        </span>
                        <span
                          className="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  
                </div>
                <div
                  className="icon secondary chevron_left"
                  onClick={[Function]}
                  style={
                    Object {
                      "left": "15px",
                    }
                  }
                >
                  ›
                </div>
              </div>
              <div
                className="gap"
              />
            </header>
            <table>
              <thead
                className="header-name"
              >
                <h6
                  className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
                >
                  Itinerary
                </h6>
              </thead>
            </table>
          </div>
          <div
            className="outer"
          >
            <table
              className="offset"
            >
              <tbody>
                <tr />
                <div
                  id="intervals"
                >
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        0:00
                      </td>
                      <td
                        className="Interval"
                        id="0:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="0:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        1:00
                      </td>
                      <td
                        className="Interval"
                        id="1:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="1:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        2:00
                      </td>
                      <td
                        className="Interval"
                        id="2:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="2:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        3:00
                      </td>
                      <td
                        className="Interval"
                        id="3:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="3:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        4:00
                      </td>
                      <td
                        className="Interval"
                        id="4:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="4:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        5:00
                      </td>
                      <td
                        className="Interval"
                        id="5:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="5:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        6:00
                      </td>
                      <td
                        className="Interval"
                        id="6:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="6:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        7:00
                      </td>
                      <td
                        className="Interval"
                        id="7:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="7:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        8:00
                      </td>
                      <td
                        className="Interval"
                        id="8:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="8:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        9:00
                      </td>
                      <td
                        className="Interval"
                        id="9:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="9:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        10:00
                      </td>
                      <td
                        className="Interval"
                        id="10:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="10:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        11:00
                      </td>
                      <td
                        className="Interval"
                        id="11:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="11:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        12:00
                      </td>
                      <td
                        className="Interval"
                        id="12:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="12:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        13:00
                      </td>
                      <td
                        className="Interval"
                        id="13:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="13:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        14:00
                      </td>
                      <td
                        className="Interval"
                        id="14:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="14:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        15:00
                      </td>
                      <td
                        className="Interval"
                        id="15:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="15:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        16:00
                      </td>
                      <td
                        className="Interval"
                        id="16:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="16:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        17:00
                      </td>
                      <td
                        className="Interval"
                        id="17:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="17:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        18:00
                      </td>
                      <td
                        className="Interval"
                        id="18:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="18:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        19:00
                      </td>
                      <td
                        className="Interval"
                        id="19:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="19:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        20:00
                      </td>
                      <td
                        className="Interval"
                        id="20:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="20:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        21:00
                      </td>
                      <td
                        className="Interval"
                        id="21:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="21:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        22:00
                      </td>
                      <td
                        className="Interval"
                        id="22:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="22:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                  <div
                    className="OneHourInterval"
                  >
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      >
                        23:00
                      </td>
                      <td
                        className="Interval"
                        id="23:00"
                        onClick={[Function]}
                      />
                    </tr>
                    <tr
                      className="OneHourInterval"
                    >
                      <td
                        className="headcol"
                      />
                      <td
                        className="Interval"
                        id="23:30"
                        onClick={[Function]}
                      />
                    </tr>
                  </div>
                </div>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
      id="unfinalized-component"
    >
      <div>
        <div>
          <header
            className="header"
          >
            <div
              className="calendar__title"
            >
              <div
                className="dropdown"
              >
                <button
                  className="dropbtn"
                  onClick={[Function]}
                >
                  Menu
                </button>
                <div
                  className="dropdown-content"
                  id="myDropdown"
                >
                  <a
                    href="#"
                  >
                    Home
                  </a>
                  <a
                    href="#"
                  >
                    Sign out
                  </a>
                </div>
              </div>
              <div>
                <button
                  className="secondary"
                  onClick={[Function]}
                  style={
                    Object {
                      "left": "165px",
                      "position": "absolute",
                      "top": "16px",
                    }
                  }
                >
                  Trip setting
                </button>
                <div
                  aria-hidden={true}
                  className="MuiBackdrop-root makeStyles-backdrop-1"
                  style={
                    Object {
                      "opacity": 0,
                      "visibility": "hidden",
                    }
                  }
                />
              </div>
            </div>
            <div
              className="gap"
            />
          </header>
          <table>
            <thead
              className="header-name"
            >
              <h6
                className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
              >
                Ideas
              </h6>
            </thead>
          </table>
        </div>
        <div
          className="MuiGrid-root"
          style={
            Object {
              "position": "absolute",
              "top": "120px",
              "width": "300px",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root"
    id="add-button-component"
  >
    <div>
      <button
        aria-label="add"
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <div />
    </div>
  </div>
</div>
`;
